library/src/main/java/com/vinted/coper/CoperImpl.kt

```diff
@@ -59,6 +72,52 @@ internal class CoperImpl(private val fragmentManager: FragmentManager) : Coper {
         }
     }
 
+    internal suspend fun getFragmentSafely(): CoperFragment {
+        val deferred = CompletableDeferred<CoperFragment>()
+        val fragment = getAttachedFragment()
+        fragment.lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                deferred.complete(fragment)
+            }
+        })
+        return deferred.await()
+    }
+
+    private suspend fun getAttachedFragment(): CoperFragment {
+        return fragmentInitializationMutex.withLock {
+            suspendCancellableCoroutine { continuation ->
+                val fragment = fragmentManager
+                    .findFragmentByTag(FRAGMENT_TAG) as? CoperFragment
+                if (fragment == null) {
+                    val newFragment = CoperFragment()
+                    if (lifecycle != null) {
+                        lifecycle.addObserver(object : DefaultLifecycleObserver {
+                            override fun onCreate(owner: LifecycleOwner) {
+                                attachFragment(newFragment, continuation::resume)
+                            }
+                        })
```

Comment:
Same here. It is not true memory leak as `CoperFragment` is tightly coupled with CoperImpl, but still, you should remove listened after you do not need it anymore